src/main/webapp/js/date.js

```diff
@@ -321,4 +308,39 @@ function parseDate(val) {
 		}
 	}
 	return null;
-}
\ No newline at end of file
+}
+
+/**
+ * Returns the timezone (in string) from a JS date object.
+ */
+function getTimeZoneStringFromDate(date) {
+    var timeZone = /\([A-Za-z0-9 ]+\)$/.exec(date.toString());
+    return timeZone[0];
+}
+
+function convertToClientTimeZone() {
```

Comment:
Header comment? 
Shouldn't this function take parameter(s)? Prefer stateless functions.